docs/guide/ethereum-provider.md

```diff
@@ -1,103 +1,164 @@
+} else {
+  console.log('Please install MetaMask!');
 }
 ```
 
-The provider API itself is very simple, and wraps [Ethereum JSON-RPC](./rpc-api) formatted messages, which is why developers usually use a convenience library for interacting with the provider,
-like [web3](https://www.npmjs.com/package/web3), [ethers](https://www.npmjs.com/package/ethers), [truffle](https://www.trufflesuite.com/), [Embark](https://framework.embarklabs.io/), or others.
-From those tools, you can generally find sufficient documentation to interact with the provider, without reading this lower-level API.
+::: warning
+In Q3 and Q4 of 2020, we are making changes to our provider API that will be breaking for some web3 sites.
 
-However, for developers of convenience libraries, and for developers who would like to use features that are not yet supported by their favorite libraries, knowledge of the provider API is essential.
+Please read the [Upcoming Breaking Changes section](#upcoming-breaking-changes) for details.
+:::
 
-[[toc]]
+The provider API is specified by [EIP-1193](https://eips.ethereum.org/EIPS/eip-1193), and is designed to be minimal.
+We recommend that all web3 site developers read the [Basic Usage section](#basic-usage).
 
-## Upcoming Provider Changes
+## Table of Contents
 
-In **Q4 2020**, we will be making some breaking changes to the provider API. The biggest change is that we will stop injecting `window.web3`. We encourage you to
-[read more about this here](https://github.com/MetaMask/metamask-extension/issues/8077).
+[[toc]]
 
-## List of Chain and Network IDs
+## Upcoming Breaking Changes
 
 ::: tip
-The chain and network IDs of these chains are the same.
+These changes are _upcoming._ Follow [this GitHub issue](https://github.com/MetaMask/metamask-extension/issues/8077) for details.
 
-Network IDs are decimal strings, while chain IDs are hexadecimal strings.
+If you are new to using the provider, or use [ethers](https://www.npmjs.com/package/ethers), you do not have to worry about these changes, and can skip ahead [to the next section](#api).
 :::
 
-```javascript
-'1': Ethereum Main Network
-'2': Morden Test network
-'3': Ropsten Test Network
-'4': Rinkeby Test Network
-'5': Goerli Test Network
-'42': Kovan Test Network
-```
+### `window.ethereum` API Changes
 
-## Properties
+In **Q3 2020** (date TBD), we are introducing some breaking changes to this API, which we encourage you to
+[read more about here](https://medium.com/metamask/breaking-changes-to-the-metamask-inpage-provider-b4dde069dd0a).
 
-### ethereum.isMetaMask
+At that time, we will:
 
-Returns `true` or `false`, representing whether the user has MetaMask installed.
+- Stop emitting `chainIdChanged`, and instead emit `chainChanged`
+- Ensure that all `chainId` values are **not** 0-prefixed
+  - For example, instead of `0x01`, we will always return `0x1`, wherever the chain ID is returned or accessible.
+- Remove the following experimental methods:
+  - `ethereum._metamask.isEnabled`
```

Comment:
Why are we removing this? What would recommend as its replacement?

Reply:
> Why are we removing this?

Several reasons!

- It's sync, but the state is referring to is async
- It refers to a concept called "enabled", which we've done away with
- Its purpose is to keep track of whether accounts are available, but there's `eth_accounts` and `accountsChanged` for that

> What would we recommend as its replacement?

The snippet in "Using the Provider", which is what I advise in 8b8a693